### PR TITLE
Cmake use interface lib

### PR DESCRIPTION
**Pull request by mirelachirica**
Changed to be in line with rest of libraries.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -50,217 +50,157 @@ add_library(nanostackeventloop STATIC)
 add_library(nanostackeventloopInterface INTERFACE)
 target_include_directories(nanostackeventloopInterface INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/nanostack-event-loop)
 
-include_directories(${NANOSTACK_LIBSERVICE_INCLUDE_DIR})
-
 target_sources(nanostackeventloop PRIVATE
     source/event.c
     source/event.h
     source/nanostack_eventloop.c
+    source/nanostack_eventloop.h
```
Comment (jarvte) [inline CMakeLists.txt:57]: Why are header-files added to nanostackeventloop sources?

Comment (mirelachirica) [reply CMakeLists.txt:57]: I was reverting changes from: https://github.com/PelionIoT/sal-stack-nanostack-eventloop/pull/51/files

But indeed those headers shouldnt be needed.

**Comment by mirelachirica** [comment]
I dont see reason for failure to run unit tests:

https://iot-mesh.isgtesting.com/job/sal_stack_nanostack_eventloop_PR/62/consoleFull

run_unit_tests.sh should be fine and running "make all test" instead of "make check"

@artokin could you take a look?

**Comment by iot-mesh** [comment]
### HW memory statistics
#### mbed-mesh-test-application, DISCO_F769NI, mbed_app_nanostack_full_ATMEL.json
| Memory      | Base                     | PR                     | Delta |
| ----------- | ------------------------ | ---------------------- | ------------------------ |
| Total Flash | 855964 | 855964 | 0 |
| Static Ram  | 122456 | 122456 | 0 |
***
Tested with commit: 0de024f4f8dba07758eed28c1fb136d5053ec2c0

File: test/stubs/ns_list_stub.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2015 ARM Limited. All rights reserved.
```
Comment (artokin) [inline test/stubs/ns_list_stub.c:2]: Would you please fix the copyright to be similar as in other files?

**Comment by iot-mesh** [comment]
### HW memory statistics
#### mbed-mesh-test-application, DISCO_F769NI, mbed_app_nanostack_full_ATMEL.json
| Memory      | Base                     | PR                     | Delta |
| ----------- | ------------------------ | ---------------------- | ------------------------ |
| Total Flash | 855964 | 855964 | 0 |
| Static Ram  | 122456 | 122456 | 0 |
***
Tested with commit: 05466c4042717e76b8336c5733582ba751a92c12

**Review by jarvte — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -50,217 +50,153 @@ add_library(nanostackeventloop STATIC)
...
+    include(GoogleTest)
+    gtest_discover_tests(ns_timer_test EXTRA_ARGS --gtest_output=xml: XML_OUTPUT_DIR nstimer_xml)
+    gtest_discover_tests(nanostack_eventloop_test EXTRA_ARGS --gtest_output=xml: XML_OUTPUT_DIR nanostack_eventloop_xml)
+    gtest_discover_tests(nanostack_eventloop_test2 EXTRA_ARGS --gtest_output=xml: XML_OUTPUT_DIR nanostack_eventloop2_xml)
+    gtest_discover_tests(event_test EXTRA_ARGS --gtest_output=xml: XML_OUTPUT_DIR event_xml)
+    gtest_discover_tests(ns_timeout_test EXTRA_ARGS --gtest_output=xml: XML_OUTPUT_DIR ns_timeout_xml)
+    gtest_discover_tests(system_timer_test EXTRA_ARGS --gtest_output=xml: XML_OUTPUT_DIR system_timer_xml)
+    if (enable_coverage_data AND ${CMAKE_PROJECT_NAME} STREQUAL "nanostackeventloop")
+        file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/html")
+        add_test(NAME coap_cov WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
```
Comment (artokin) [inline CMakeLists.txt:197]: Would you please change the test name `coap_cov` to be line `event_loop_cov`

**Review by artokin — APPROVED** [review]

**Comment by iot-mesh** [comment]
### HW memory statistics
#### mbed-mesh-test-application, DISCO_F769NI, mbed_app_nanostack_full_ATMEL.json
| Memory      | Base                     | PR                     | Delta |
| ----------- | ------------------------ | ---------------------- | ------------------------ |
| Total Flash | 855964 | 855964 | 0 |
| Static Ram  | 122456 | 122456 | 0 |
***
Tested with commit: 1c2c488eace39ef57d78a74a915a2df129f64ce8

**Comment by iot-mesh** [comment]
### HW memory statistics
#### mbed-mesh-test-application, DISCO_F769NI, mbed_app_nanostack_full_ATMEL.json
| Memory      | Base                     | PR                     | Delta |
| ----------- | ------------------------ | ---------------------- | ------------------------ |
| Total Flash | 855964 | 855964 | 0 |
| Static Ram  | 122456 | 122456 | 0 |
***
Tested with commit: 7019b69667ab306f1f85bc20c13bd040bc5a92fc

**Review by artokin — APPROVED** [review]

**Pull request merged**